hack/create-kind-cluster.sh

```diff
@@ -1,197 +1,42 @@
+if [ "${running}" != 'true' ]; then
+  docker run \
+    -d --restart=always -p "${reg_port}:5000" --name "${reg_name}" \
+    registry:2
+fi
 
-function insecure-registry-config-cmd() {
-  local config_file="${1}"
-  case $config_file in
-    $docker_daemon_config)
-      echo "cat <<EOF > ${docker_daemon_config}
-{
-    \"insecure-registries\": [\"${CONTAINER_REGISTRY_HOST}\"]
-}
+# create a cluster with the local registry enabled in containerd
+cat <<EOF | kind create cluster --name ${cluster_name} --config=-
+kind: Cluster
+apiVersion: kind.x-k8s.io/v1alpha4
+containerdConfigPatches:
+- |-
+  [plugins."io.containerd.grpc.v1.cri".registry.mirrors."localhost:${reg_port}"]
+    endpoint = ["http://${reg_name}:${reg_port}"]
 EOF
-"
-      ;;
-    $containerd_config)
-     echo "sed -i '/\[plugins.cri.registry.mirrors\]/a [plugins.cri.registry.mirrors."\"${CONTAINER_REGISTRY_HOST}\""]\nendpoint = ["\"http://${CONTAINER_REGISTRY_HOST}\""]' ${containerd_config}"
-     ;;
-    *)
-     echo "Sorry, config insecure registy is not supported for $config_file"
-     ;;
-  esac
-}
-
-
-# wait-for-condition blocks until the provided condition becomes true
-#
-# Globals:
-#  None
-# Arguments:
-#  - 1: message indicating what conditions is being waited for (e.g. 'config to be written')
-#  - 2: a string representing an eval'able condition.  When eval'd it should not output
-#       anything to stdout or stderr.
-#  - 3: optional timeout in seconds.  If not provided, waits forever.
-# Returns:
-#  1 if the condition is not met before the timeout
-function wait-for-condition() {
-  local msg=$1
-  # condition should be a string that can be eval'd.
-  local condition=$2
-  local timeout=${3:-}
-
-  local start_msg="Waiting for ${msg}"
-  local error_msg="[ERROR] Timeout waiting for ${msg}"
-
-  local counter=0
-  while ! eval ${condition}; do
-    if [[ "${counter}" = "0" ]]; then
-      echo -n "${start_msg}"
-    fi
 
-    if [[ -z "${timeout}" || "${counter}" -lt "${timeout}" ]]; then
-      counter=$((counter + 1))
-      if [[ -n "${timeout}" ]]; then
-        echo -n '.'
-      fi
-      sleep 1
-    else
-      echo -e "\n${error_msg}"
-      return 1
-    fi
-  done
+# connect the registry to the cluster network
+docker network disconnect "kind" "${reg_name}" || true
+docker network connect "kind" "${reg_name}"
 
-  if [[ "${counter}" != "0" && -n "${timeout}" ]]; then
-    echo ' done'
-  fi
-}
-readonly -f wait-for-condition
-
-echo "Creating cluster $CLUSTER_NAME"
-create-cluster $CLUSTER_NAME
-
-echo "Complete"
-
-if [[ ! "${OVERWRITE_KUBECONFIG}" ]]; then
-    echo <<EOF "OVERWRITE_KUBECONFIG was not set so ${kubeconfig} was not modified. \
-You can access your clusters by setting your KUBECONFIG environment variable using:
-
-export KUBECONFIG=\"${KUBECONFIG}\"
-
-Then you can overwrite ${kubeconfig} if you prefer using:
-
-kubectl config view --flatten > ${kubeconfig}
-unset KUBECONFIG
-"
-EOF
-fi
+# tell https://tilt.dev to use the registry
```

Comment:
Is this section required? If so, we should remove the references to tilt.

Reply:
Didn't even realize, no it does not seem necessary, removing.